homeassistant/components/acaia/binary_sensor.py

```diff
@@ -0,0 +1,73 @@
+"""Binary sensor platform for Acaia scales."""
+
+from collections.abc import Callable
+from dataclasses import dataclass
+
+from aioacaia.acaiascale import AcaiaScale
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDeviceClass,
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .coordinator import AcaiaConfigEntry
+from .entity import AcaiaEntity
+
+
+@dataclass(kw_only=True, frozen=True)
+class AcaiaBinarySensorEntityDescription(BinarySensorEntityDescription):
+    """Description for Acaia binary sensor entities."""
+
+    is_on_fn: Callable[[AcaiaScale], bool]
+    available_fn: Callable[[AcaiaScale], bool] | None = None
+
+
+BINARY_SENSORS: tuple[AcaiaBinarySensorEntityDescription, ...] = (
+    AcaiaBinarySensorEntityDescription(
+        key="timer_running",
+        translation_key="timer_running",
+        device_class=BinarySensorDeviceClass.RUNNING,
+        is_on_fn=lambda scale: scale.timer_running,
+    ),
+    AcaiaBinarySensorEntityDescription(
+        key="connected",
+        translation_key="connected",
+        device_class=BinarySensorDeviceClass.CONNECTIVITY,
+        is_on_fn=lambda scale: scale.connected,
+        available_fn=lambda _: True,
```

Comment:
This seems wrong with the logic there. THis means connected is handled differently compared to what the coordinator reports. However, if the device/service isn't available, we can possibly tell if one is connected or not.

I think this is not a good sensor, as the state of the entities already provide this information?

Reply:
It's easier to react on a sensor that tells you what's going on than on guessing based on unavailable states. Also imo that's the purpose of the device class.

Reply:
I remember why I added that. I have a RestoreSensor that's of course also available if the device is not available, so I wanted to make it clear if the device is connected. But if you think I should remove it I will.